File: src/main/java/io/github/oleksivio/telegram/bot/core/controller/annotations/parser/unpacker/ChosenInlineResultUnpackerParser.java
package io.github.oleksivio.telegram.bot.core.controller.annotations.parser.unpacker;

import io.github.oleksivio.telegram.bot.core.controller.handler.unpack.UpdateUnpacker;
import org.springframework.stereotype.Component;
import io.github.oleksivio.telegram.bot.api.annotations.receiver.ChosenInlineResultReceiver;
import io.github.oleksivio.telegram.bot.api.model.objects.inline.ChosenInlineResult;

import java.util.Optional;

@Component
public class ChosenInlineResultUnpackerParser extends UnpackerParser<ChosenInlineResultReceiver, ChosenInlineResult> {

    @Override
    public Class<ChosenInlineResultReceiver> getAnnotationClass() {
        return ChosenInlineResultReceiver.class;
    }

    @Override
    public UpdateUnpacker<ChosenInlineResult> getUpdateUnpacker() {
        return new UpdateUnpacker<>(in -> Optional.ofNullable(in.getChosenInlineResult()), ChosenInlineResult.class);
    }
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/core/controller/annotations/parser/unpacker/ChosenInlineResultUnpackerParser.kt
package io.github.oleksivio.telegram.bot.core.controller.annotations.parser.unpacker

import io.github.oleksivio.telegram.bot.api.annotations.receiver.ChosenInlineResultReceiver
import io.github.oleksivio.telegram.bot.api.model.objects.inline.ChosenInlineResult
import io.github.oleksivio.telegram.bot.core.controller.handler.unpack.UpdateUnpacker
import org.springframework.stereotype.Component

@Component
class ChosenInlineResultUnpackerParser : UnpackerParser<ChosenInlineResultReceiver, ChosenInlineResult>() {
    override val updateUnpacker: UpdateUnpacker<ChosenInlineResult>
        get() = UpdateUnpacker({ it.chosenInlineResult }, ChosenInlineResult::class)

    override val parserAnnotationClass = ChosenInlineResultReceiver::class

}
